Applications/SimpleShot/layers/centering.h

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   centering.h
+ * @date   08 Jan 2021
+ * @brief  This file contains the simple centering layer which has hardcoded
+ * mean feature vectors from given combinations
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __CENTERING_H__
+#define __CENTERING_H__
+#include <string>
+
+/// @todo migrate these to API
+#include <layer_internal.h>
+#include <manager.h>
+
+#include <tensor.h>
+
+namespace simpleshot {
+namespace layers {
+
+/// @todo inherit this to API
+// class CenteringLayer : public ml::train::Layer {
+class CenteringLayer : public nntrainer::Layer {
+public:
+  /**
+   * @brief Construct a new Centering Layer object that does elementwise
+   * subtraction from mean feature vector
+   */
+  CenteringLayer() : Layer() {}
+
+  /**
+   * @brief Construct a new Centering Layer object
+   *
+   * @param feature_path feature path to read the variable
+   */
+  CenteringLayer(const std::string &feature_path);
+
+  /**
+   *  @brief  Move constructor.
+   *  @param[in] CenteringLayer &&
+   */
+  CenteringLayer(CenteringLayer &&rhs) noexcept = default;
+
+  /**
+   * @brief  Move assignment operator.
+   * @parma[in] rhs CenteringLayer to be moved.
+   */
+  CenteringLayer &operator=(CenteringLayer &&rhs) = default;
+
+  /**
+   * @brief Destroy the Centering Layer object
+   *
+   */
+  ~CenteringLayer() {}
+
+  using nntrainer::Layer::setProperty;
+
+  /**
+   * @brief     set Property of layer,
+   * feature_path: feature *.bin that contains mean feature vector that will be
+   * used for the model.
+   * @param[in] values values of property
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  int setProperty(std::vector<std::string> values) override;
+
+  /**
+   * @brief initializing nntrainer
+   *
+   * @return int ML_ERROR_NONE if success
+   */
+  int initialize(nntrainer::Manager &manager) override;
+
+  /**
+   * @brief nntrainer forwarding function
+   */
+  void forwarding(bool training = true) override;
+
+  /**
+   * @brief     calc the derivative to be passed to the previous layer
+   */
+  void calcDerivative() override;
+
+  /**
+   * @brief Get the Type object
+   *
+   * @return const std::string
+   */
+  const std::string getType() const override { return CenteringLayer::type; }
+
+  static const std::string type;
+
+private:
+  std::string feature_path;
+  nntrainer::Tensor mean_feature_vector;
```

Comment:
let's use weight with trainable as false?

Reply:
I am assuming this could migrate using only api. So that can be done but delayed until the api is fixed. Afte we fixate the api, I am going to migrate examples.

Reply:
ok.